server/src/main/java/com/cloud/api/ApiDBUtils.java

```diff
@@ -1568,7 +1568,7 @@ public static GuestOSCategoryVO findGuestOsCategoryById(long catId) {
     }
 
     public static VpcVO findVpcById(long vpcId) {
-        return s_vpcDao.findById(vpcId);
+        return s_vpcDao.findByIdIncludingRemoved(vpcId);
```

Comment:
Could this have side effects? You may need to check, or instead add a new method that finds VPC by ID including removed?

Reply:
Can you please tell me why this will have side effects?

Reply:
@ravening it would be better to add new method in ApiDBUtils.java

Reply:
ping @ravening ^^

Reply:
@rhtyd fixed it. added a new function